src/test/scala/streamer/DataWriterTest.scala

```diff
@@ -0,0 +1,54 @@
+package streamer
+
+import chisel3._
+import org.scalatest.flatspec.AnyFlatSpec
+import chiseltest._
+import scala.math.BigInt
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.Tag
+import scala.util.control.Breaks._
+
+// data writer initial test
+class DataWriterTest
+    extends AnyFlatSpec
+    with ChiselScalatestTester
+    with Matchers {
+  "DUT" should "pass" in {
+    test(new DataWriter).withAnnotations(
+      Seq(WriteVcdAnnotation)
+    ) { dut =>
+      dut.clock.step(5)
+      // give valid config, give the proper unrolling strides so that is a aligned in one TCDM bank
+      dut.io.unrollingStrides_csr_i.bits(0).poke(1)
+      dut.io.unrollingStrides_csr_i.bits(1).poke(8)
+      dut.io.unrollingStrides_csr_i.valid.poke(1)
+      dut.io.data_fifo_i.valid.poke(1)
+      for (i <- 0 until DataWriterTestParameters.dataWriterTcdmPorts) {
+        dut.io.tcdm_ready_i(i).poke(1)
+      }
+      dut.clock.step(5)
+      dut.io.unrollingStrides_csr_i.valid.poke(0)
+
+      dut.io.ptr_agu_i.bits.poke(100)
+      dut.io.ptr_agu_i.valid.poke(1)
+
+      // wait for getting the enough address
+      var ready_counter = 0
+      breakable {
+        while (true) {
+          if (dut.io.ptr_agu_i.ready.peekBoolean()) {
+            ready_counter = ready_counter + 1
+            dut.clock.step(5)
+            if (ready_counter == 8) {
+              break()
+            }
+          }
+        }
+      }
```

Comment:
What is being tested here? Add comments!

Reply:
There is nothing to be tested. It just waits until all the addresses are generated and then finishes the simulation...